bundles/org.openhab.binding.mideaac/src/main/java/org/openhab/binding/mideaac/internal/connection/ConnectionManager.java

```diff
@@ -0,0 +1,540 @@
+    private boolean deviceIsConnected;
+    private int droppedCommands = 0;
+
+    /**
+     * True allows command retry if null response
+     */
+    private boolean retry = true;
+
+    public ConnectionManager(String ipAddress, int ipPort, int timeout, String key, String token, String cloud,
+            String email, String password, String deviceId, int version, boolean promptTone) {
+        this.deviceIsConnected = false;
+        this.ipAddress = ipAddress;
+        this.ipPort = ipPort;
+        this.timeout = timeout;
+        this.key = key;
+        this.token = token;
+        this.cloud = cloud;
+        this.deviceId = deviceId;
+        this.version = version;
+        this.promptTone = promptTone;
+        this.lastResponse = new Response(HexFormat.of().parseHex("C00042667F7F003C0000046066000000000000000000F9ECDB"),
+                version, "query", (byte) 0xc0);
+        this.cloudProvider = CloudProviderDTO.getCloudProvider(cloud);
+        this.security = new Security(cloudProvider);
+    }
+
+    private Socket socket = new Socket();
+    private InputStream inputStream = new ByteArrayInputStream(new byte[0]);
+    private DataOutputStream writer = new DataOutputStream(System.out);
+
+    /**
+     * Gets last response
+     * 
+     * @return byte array of last response
+     */
+    public Response getLastResponse() {
+        return this.lastResponse;
+    }
+
+    /**
+     * Validate if String is blank
+     * 
+     * @param str string to be evaluated
+     * @return boolean true or false
+     */
+    public static boolean isBlank(String str) {
+        return str.trim().isEmpty();
+    }
+
+    /**
+     * After checking if the key and token need to be updated (Default = 0 Never)
+     * The socket is established with the writer and inputStream (for reading responses)
+     * The device is considered connected. V2 devices will proceed to send the poll or the
+     * set command. V3 devices will proceed to authenticate
+     */
+    public synchronized void connect() throws MideaConnectionException, MideaAuthenticationException {
+        logger.trace("Connecting to {}:{}", ipAddress, ipPort);
+
+        int maxTries = 3;
+        int retryCount = 0;
+
+        // Open socket
+        // Retry addresses most common wifi connection problems- wait 5 seconds and try again
+        while (retryCount < maxTries) {
+            try {
+                socket = new Socket();
+                socket.setSoTimeout(timeout * 1000);
+                socket.connect(new InetSocketAddress(ipAddress, ipPort), timeout * 1000);
+                break;
+            } catch (IOException e) {
+                retryCount++;
+                if (retryCount < maxTries) {
+                    try {
+                        Thread.sleep(5000);
+                    } catch (InterruptedException ex) {
+                        logger.debug("An interupted error (socket retry) has occured {}", ex.getMessage());
```

Comment:
Stop processing and exit immediate when an InterruptException occurs.

Reply:
As I recall that won't work as I intended (with 3 tries).  The exceptions aren't serious and ephemeral in nature.  I haven't had that INFO log message in months (after I went from 2 to three tries.  That's why I forgot it was there.

Reply:
I have separated the catches